### PR TITLE
Anko - Change Group layout to anko

### PR DESCRIPTION
**Pull request by at-haingo**
#### Related issues (Required)
- Task #111 

#### Description (Required)
- [x] Create GroupInfoFragmentUI
- [x] Create CreateGroupFragmentUI
- [x] Create GroupActivityUI
- [x] Create NonGroupFragmentUI
- [x] Create ReloadFragmentUI
- [x] Create ItemGroupMemberUI

### CODE REVIEW

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import android.view.View
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ * Fragment group user
+ * Created by haingoq on 28/11/2017.
+ */
+class GroupInfoFragment: BaseFragment() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        return GroupInfoFragmentUI().createView(AnkoContext.Companion.create(context, this))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/group/UITest.kt [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package vn.asiantech.way.ui.group
+
+import android.os.Bundle
+import vn.asiantech.way.R
+import vn.asiantech.way.ui.base.BaseActivity
+
+/**
+ * Created by haingoq on 28/11/2017.
+ */
+class UITest: BaseActivity() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/UITest.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                rightPadding = padding
+            }.lparams(matchParent, dimen(R.dimen.toolbar_height))
+
+            textViewInfo()
+            textViewInfo()
+            textViewInfo(resources.getString(R.string.members_list))
+
+            relativeLayout {
+                lparams(matchParent, wrapContent)
+                padding = dip(10)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: WARNING: Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers.

File: app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ * Created by haingoq on 29/11/2017.
+ */
+class ReloadFragment: BaseFragment() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        return ReloadFragmentUI().createView(AnkoContext.create(context, this))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: WARNING: Checks whether files end with a line separator.

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+import android.view.View
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ *
+ * Created by haingoq on 29/11/2017.
+ */
+class NonGroupFragment: BaseFragment() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt]: WARNING:

Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt]: WARNING: Unexpected indentation

File: app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+        return NonGroupFragmentUI().createView(AnkoContext.create(context, this))
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt]: WARNING: Line detected that is longer than the defined maximum line length in the code style.

File: app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+
+import android.os.Bundle
+import org.jetbrains.anko.setContentView
+import vn.asiantech.way.ui.base.BaseActivity
+
+/**
+ *
+ * Created by haingoq on 29/11/2017.
+ */
+class GroupActivity: BaseActivity() {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/ItemGroupMemberUI.kt [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+            }.lparams(dimen(R.dimen.group_screen_avatar_width),
+                    dimen(R.dimen.group_screen_avatar_width))
+        }
+    }
+
+    /*
+     * Add circleImageView library
+     */
+    private inline fun ViewManager.circleImageView(theme: Int = 0, init: CircleImageView.() -> Unit):
+            CircleImageView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ItemGroupMemberUI.kt]: WARNING:

File: app/src/main/java/vn/asiantech/way/ui/group/CreateGroupFragmentUI.kt [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    internal lateinit var edtGroupName: EditText
+    internal lateinit var btnCreateGroup: Button
+    internal lateinit var btnBack: Button
+
+    override fun createView(ui: AnkoContext<CreateGroupFragment>) = with(ui) {
+        verticalLayout {
+            lparams(matchParent, matchParent)
+            backgroundColor = ActivityCompat.getColor(context, R.color.colorBlueLight)
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/CreateGroupFragmentUI.kt]: use `backgroundResource = R.color.colorBlueLight`

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+        private const val ID_IMG_INVITE = 1001
+        private const val MAX_LINES = 1
+    }
+
+    internal lateinit var tvGroupName: TextView
+    internal lateinit var tvMembersCount: TextView
+    internal lateinit var tvCreateAt: TextView
+    internal lateinit var imgInvite: ImageView
+    internal lateinit var imgLeave: ImageView
+    internal lateinit var recyclerViewMembers: RecyclerView
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: `recyclerViewMembers` change to `recyclerViewMember`

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+            tvCreateAt = textViewInfo()
+            textViewInfo(resources.getString(R.string.members_list))
+
+            relativeLayout {
+                lparams(matchParent, wrapContent)
+                padding = dimen(R.dimen.group_screen_tv_count_padding_left)
+
+                imgInvite = imageView {
+                    id = ID_IMG_INVITE
+                    backgroundResource = R.drawable.ic_person_add_deep_purple_a200_36dp
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+                imgLeave = imageView {
+                    backgroundResource = R.drawable.ic_exit_to_app_deep_purple_a200_36dp
+                }.lparams {
+                    leftMargin = dimen(R.dimen.group_screen_img_leave_margin)
+                    rightOf(ID_IMG_INVITE)
+                }
+            }
+
+            swipeRefreshLayout {
+                recyclerViewMembers = recyclerView {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: add `layoutmanager`

File: app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragmentUI.kt [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+import org.jetbrains.anko.*
+import vn.asiantech.way.R
+
+/**
+ *
+ * Created by haingoq on 29/11/2017.
+ */
+class NonGroupFragmentUI : AnkoComponent<NonGroupFragment> {
+    internal lateinit var btnCreateGroup: Button
+    internal lateinit var btnViewInvites: Button
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragmentUI.kt]: `btnViewInvites` change to `btnViewInvite`

File: app/src/main/java/vn/asiantech/way/ui/group/ReloadFragmentUI.kt [hunk 19]
```diff
@@ -0,0 +1,35 @@
...
+
+            textView(R.string.touch_to_reload) {
+                textColor = Color.BLACK
+                textSize = px2dip(dimen(R.dimen.notification_size))
+            }.lparams {
+                gravity = Gravity.CENTER
+            }
+
+            imgReload = imageView {
+                backgroundResource = R.drawable.ic_replay_indigo_300_48dp
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragmentUI.kt]: remove this line

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 20]
```diff
@@ -0,0 +1,89 @@
+package vn.asiantech.way.ui.group
+
+import android.graphics.Color
+import android.support.v4.app.ActivityCompat
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: WARNING: Unused import

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finished review this PR, please check it again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt [hunk 21]
```diff
@@ -0,0 +1,16 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        GroupActivityUI().setContentView(this)
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupActivity.kt]: Please create `ui` variable

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ * Fragment group user
+ * Created by haingoq on 28/11/2017.
+ */
+class GroupInfoFragment : BaseFragment() {
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragment.kt]: please create ui variable

File: app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt [hunk 23]
```diff
@@ -0,0 +1,88 @@
...
+import org.jetbrains.anko.support.v4.swipeRefreshLayout
+import vn.asiantech.way.R
+
+/**
+ *
+ * Created by haingoq on 28/11/2017.
+ */
+class GroupInfoFragmentUI : AnkoComponent<GroupInfoFragment> {
+    companion object {
+        private const val ID_IMG_INVITE = 1001
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/GroupInfoFragmentUI.kt]: please move this id to resource

File: app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ *
+ * Created by haingoq on 29/11/2017.
+ */
+class NonGroupFragment : BaseFragment() {
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/NonGroupFragment.kt]: please create `ui` variable

File: app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+import android.view.ViewGroup
+import org.jetbrains.anko.AnkoContext
+import vn.asiantech.way.ui.base.BaseFragment
+
+/**
+ *
+ * Created by haingoq on 29/11/2017.
+ */
+class ReloadFragment : BaseFragment() {
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
```
Comment (at-tienhoang) [inline app/src/main/java/vn/asiantech/way/ui/group/ReloadFragment.kt]: Please create ui variable

File: app/src/main/java/vn/asiantech/way/ui/group/ItemGroupMemberUI.kt [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+                    .lparams(dimen(R.dimen.group_screen_avatar_width),
+                            dimen(R.dimen.group_screen_avatar_width))
+        }
+    }
+
+    /*
+     * Add circleImageView library
+     */
+    private inline fun ViewManager.circleImageView(theme: Int = 0, init: CircleImageView.() -> Unit):
+            CircleImageView {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/ItemGroupMemberUI.kt]: WARNING:

**Pull request merged**